controllers/ProposalController.php

```diff
@@ -277,4 +283,139 @@ private function getReviewedAndNoPublishedProposalsForAReviewer(): ActiveDataPro
         return $reviewedAndNoPublishedProposalsForAReviewer;
     }
 
+    /**
+     * Display a form to create a Proposal
+     * Save the proposal in DDB when form is submitted
+     *
+     * @return string|yii\web\Response
+     * @throws \Throwable
+     */
+    public function actionCreateProposal()
+    {
+        $model = new CreateProposalForm();
+
+        if ($model->load(Yii::$app->request->post()) && $model->validate()) {
+            $transaction = Yii::$app->db->beginTransaction();
+
+            try {
+                $post = Yii::$app->request->post();
+                $proposal = $this->saveProposal($post['CreateProposalForm']['title']);
+                $this->saveProposalContent($post['CreateProposalForm']['content'], $proposal);
+
+                if (!empty($_FILES['CreateProposalForm']['name']['relatedFile'])) {
+                    $uploadedFile = $_FILES['CreateProposalForm'];
+                    $movedFilename = $this->moveUploadedFileToServer($uploadedFile, $proposal);
+                    $this->saveProposalRelatedFile($movedFilename, $proposal);
+                }
+
+                $transaction->commit();
+                return $this->redirect('/proposal/my-proposals/' . $proposal->id);
+            } catch (CannotHandleUploadedFileException $cannotHandleUploadedFileException) {
+                $transaction->rollBack();
+                return $this->render('create-proposal', ['model' => $model, 'error' => 'Invalid file']);
+
+            } catch(\Throwable $e) {
+                $transaction->rollBack();
+                throw $e;
+            }
+        } else {
+            return $this->render('create-proposal', ['model' => $model, 'error' => null]);
+        }
+    }
+
+    /**
+     * Save proposal in DDB
+     *
+     * @param $proposalTitle
+     * @return Proposal
+     * @throws CannotSaveException
+     */
+    private function saveProposal(string $proposalTitle): Proposal
+    {
+        $proposal = new Proposal();
+        $proposal->title = $proposalTitle;
+        $proposal->submitter_id = mainController::getCurrentUser()->id;
+        $proposal->status = 'pending';
```

Comment:
I think we have to create a Proposal model which extends the base databaseModels/Proposal, create constants for each proposal status, and use the constants instead of using directly the name of the status

Reply:
Yes, you're right. In case of future change it will be easier to edit the const one time instead of several times in code.